ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcInputFormat.java

```diff
@@ -2675,12 +2676,13 @@ public static TypeDescription convertTypeInfo(TypeInfo info) {
   public static TypeDescription getDesiredRowTypeDescr(Configuration conf,
                                                        boolean isAcidRead,
                                                        int dataColumns) {
-
     String columnNameProperty = null;
     String columnTypeProperty = null;
 
     ArrayList<String> schemaEvolutionColumnNames = null;
     ArrayList<TypeDescription> schemaEvolutionTypeDescrs = null;
+    // Make sure we split colNames using the right Delimiter
+    final String columnNameDelimiter = conf.get(serdeConstants.COLUMN_NAME_DELIMITER, String.valueOf(SerDeUtils.COMMA));
```

Comment:
Hey @abstractdog  thanks for taking a look!
> this makes me think that in order to use commas in column names, you need to define another column name delimiter, otherwise those are cannot be distinguished from each other, right?
We already check this corner case when creating the https://github.com/apache/hive/blob/95f3d6512f35839f2fad3cfd608616534e506a4b/ql/src/java/org/apache/hadoop/hive/ql/plan/PlanUtils.java#L562

https://github.com/pgaref/hive/blob/0d2b39ac180d809788f662fbe3271482cd4d909d/standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreUtils.java#L1562
actually checks for commas in colNames and uses \0 to split them instead.

This PR is just making use of the custom delimiter that was forgotten for OrcInputFormat but is done for others e.g., OrcOutputFormat

Reply:
cool, thanks for clarifying, LGTM